Update custom-op repo for manylinux pip packages

yifeif opened the pull request:
- Add option and flag to configure.sh for manylinux2010 packages
- Update build instructions at README.md

yifeif commented:
cc: @ewilderj @seanpmorgan

av8ramit commented on README.md:
```diff
@@ -1,18 +1,33 @@
+This guide including example for both cpu and gpu ops.
+
+Starting from Aug 1, 2019, nightly previews `tf-nightly` and `tf-nightly-gpu`, as well as
+official releases `tensorflow` and `tensorflow-gpu` past version 1.14.1 are now built with a
```
1.14.0, we did not do a 1.14.1.

yifeif replied:
Done, thanks.

gunan commented on README.md:
````diff
@@ -124,24 +139,41 @@ cd my_op
-Next, set up a Docker container using the provided Docker image for building and testing the ops. The provided Docker images `tensorflow/tensorflow:custom-op` and `tensorflow/tensorflow:custom-op-gpu` are based on Ubuntu 14.04, and it contains the same versions of tools and libraries used for building the official TensorFlow pip packages. It also comes with Bazel pre-installed. We have seen many  ...
+Next, set up a Docker container using the provided Docker image for building and testing the ops. We provide two sets of Docker images for different versions of pip packages. If the pip package you are building against was released before Aug 1, 2019 and has manylinux1 tag, please use Docker images `tensorflow/tensorflow:custom-op` and `tensorflow/tensorflow:custom-op-gpu`, which are based on Ubu ...
 ```bash
+# For manylinux1
````
Not really manylinux1, let's just call this releases before 1.14, or releases labeled manylinux1

yifeif replied:
Updated

gunan commented on README.md:
````diff
@@ -124,24 +139,41 @@ cd my_op
+Next, set up a Docker container using the provided Docker image for building and testing the ops. We provide two sets of Docker images for different versions of pip packages. If the pip package you are building against was released before Aug 1, 2019 and has manylinux1 tag, please use Docker images `tensorflow/tensorflow:custom-op` and `tensorflow/tensorflow:custom-op-gpu`, which are based on Ubu ...
 ```bash
+# For manylinux1
 docker pull tensorflow/tensorflow:custom-op
````
Should we push these as `tensorflow/tensorflow:custom-op-ubuntu14` and change them here, preparing for the transition?

yifeif replied:
Sounds good. I pushed new images and updated the guide with the new names.

seanpmorgan commented:
One last issue. If I try auditwheel repair instead of show:
```
auditwheel repair artifacts/tensorflow_zero_out-0.0.1-cp27-cp27mu-linux_x86_64.whl 
```

It returns `auditwheel: error: cannot find the 'patchelf' tool, which is required`. So we might want to install that in the Dockerfile as well.

ewilderj commented:
cc @yongtang

yifeif commented:
Really appreciate the feedback @seanpmorgan!
I have updated the docker images with patchelf and pip, and updated the setup.py according to your comments. Let me know if it's working for you now.
I'm also setting up nightly releases for all of these docker images, so they will automatically get updated with the latest fixes.

seanpmorgan commented:
Thanks so much @yifeif !

One more fix I discovered after posting:
If you try to compile the GPU op with the installed version of bazel it will error:
`bazel test tensorflow_time_two:time_two_ops_py_test`
```
ERROR: error loading package 'tensorflow_time_two': Encountered error while reading extension file 'cuda/build_defs.bzl': no such package '@local_config_cuda//cuda': Traceback (most recent call last):
        File "/workspace/gpu/cuda_configure.bzl", line 1297
                _create_local_cuda_repository(repository_ctx)
        File "/workspace/gpu/cuda_configure.bzl", line 1285, in _create_local_cuda_repository
                _tpl(repository_ctx, "crosstool:CROSSTO...", <2 more arguments>)
        File "/workspace/gpu/cuda_configure.bzl", line 1288, in _tpl
                cuda_defines + _get_win_cuda_defines(repository_ctx)
The `+` operator for dicts is deprecated and no longer supported. Please use the `update` method instead. You can temporarily enable the `+` operator by passing the flag --incompatible_disallow_dict_plus=false
```
The fix looks something like this:
https://github.com/tensorflow/addons/pull/389/commits/6b9eb625d2af030cc7db04554c784208827f22f8


Also while we're updating the Dockerfiles we have two more requests which were present in the previous custom-op docker images
1. Install buildifer for bazel linting
```
wget https://github.com/bazelbuild/buildtools/releases/download/0.4.5/buildifier
chmod +x buildifier
mv buildifier /usr/local/bin/.
```
2. Install auditwheel 
```
pip3 install auditwheel
```

yifeif commented:
Thanks again @seanpmorgan!
I added a patch for fixing `--incompatible_disallow_dict_plus`, as well as for `--incompatible_bzl_disallow_load_after_statement`. 
Also sending another change from internal to add install_buildifier step to the docker image(should mention this PR once it gets into the main repo).

yongtang commented:
@yifeif Thanks for the PR, looks great!

yifeif commented:
Thanks all for reviewing!
